### PR TITLE
Fix node type & name inconsistency

### PR DESCRIPTION
**Pull request by ValikVityaz**

### CODE REVIEW

File: sources/plugins/scg/scgscene.cpp [hunk 1]
```diff
@@ -359,17 +359,28 @@ SCgBaseCommand* SCgScene::changeIdtfCommand(SCgObject *object, const QString &id
...
     {
         SCgAlphabet &alphabet = SCgAlphabet::getInstance();
         bool typeChanged = false;
         QStringList splittedAlias = object->typeAlias().split("/");
         if (object->type() == SCgNode::Type || object->type() == SCgPair::Type)
         {
-            if (idtf.startsWith('_') && splittedAlias.at(1) != alphabet.aliasFromConstCode(SCgAlphabet::Var))
+            bool isVar = splittedAlias.at(1) == alphabet.aliasFromConstCode(SCgAlphabet::Var);
+            bool shouldBeVar = idtf.startsWith('_');
+            if (idtf.length() != 0)
```
Comment (MikhailSadovsky) [inline sources/plugins/scg/scgscene.cpp]: This check was in line 362

File: sources/plugins/scg/scgscene.cpp [hunk 2]
```diff
@@ -409,12 +420,30 @@ SCgBaseCommand* SCgScene::changeIdtfCommand(SCgObject *object, const QString &id
...
 
     SCgBaseCommand* cmd = new SCgCommandObjectTypeChange(this, object, type, parentCmd);
 
+    QString oldIdtf = object->idtfValue();
+    if (isInitial && !oldIdtf.isEmpty()) {
+        QStringList splittedAlias = type.split("/");
+        SCgAlphabet &alphabet = SCgAlphabet::getInstance();
+        bool isTypeVar = splittedAlias.at(1) == alphabet.aliasFromConstCode(SCgAlphabet::Var);
+        bool isNameVar = oldIdtf.startsWith('_');
+        if (!oldIdtf.isEmpty())
```
Comment (MikhailSadovsky) [inline sources/plugins/scg/scgscene.cpp]: This check was in line 430

File: sources/plugins/scg/scgscene.cpp [hunk 3]
```diff
@@ -409,12 +420,30 @@ SCgBaseCommand* SCgScene::changeIdtfCommand(SCgObject *object, const QString &id
...
 
-SCgBaseCommand* SCgScene::changeObjectTypeCommand(SCgObject *object, const QString &type, SCgBaseCommand* parentCmd, bool addToStack)
+SCgBaseCommand* SCgScene::changeObjectTypeCommand(SCgObject *object, const QString &type, SCgBaseCommand* parentCmd, bool addToStack, bool isInitial)
 {
     QString oldType = object->typeAlias();
 
     SCgBaseCommand* cmd = new SCgCommandObjectTypeChange(this, object, type, parentCmd);
 
+    QString oldIdtf = object->idtfValue();
+    if (isInitial && !oldIdtf.isEmpty()) {
```
Comment (MikhailSadovsky) [inline sources/plugins/scg/scgscene.cpp]:
```suggestion
    if (isInitial && !oldIdtf.isEmpty()) 
    {
```

File: sources/plugins/scg/scgscene.cpp [hunk 4]
```diff
@@ -409,12 +420,30 @@ SCgBaseCommand* SCgScene::changeIdtfCommand(SCgObject *object, const QString &id
...
+        SCgAlphabet &alphabet = SCgAlphabet::getInstance();
+        bool isTypeVar = splittedAlias.at(1) == alphabet.aliasFromConstCode(SCgAlphabet::Var);
+        bool isNameVar = oldIdtf.startsWith('_');
+        if (!oldIdtf.isEmpty())
+        {
+            if (isTypeVar && !isNameVar) changeIdtfCommand(object, oldIdtf.insert(0, '_'), cmd, false);
+            if (!isTypeVar && isNameVar)
+            {
+                int i = 0;
+                while (oldIdtf[i] == '_') ++i;
```
Comment (MikhailSadovsky) [inline sources/plugins/scg/scgscene.cpp]: Could you give an example why we need this cycle?

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

**Comment by MikhailSadovsky** [comment]
Please specify the issue this change refers to

File: sources/plugins/scg/scgscene.cpp (hunk 4 above)
Comment (ValikVityaz) [reply sources/plugins/scg/scgscene.cpp]: If for some reason the name of the node begins with multiple underscores (e.g. ___a)

**Review by MikhailSadovsky — APPROVED** [review]

**Review by Zioba — APPROVED** [review]

**Comment by deniskoronchik** [comment]
What is an issue that is fixed by this PR?

**Comment by deniskoronchik** [comment]
As mentioned in the [issue](https://github.com/ostis-dev/kbe/issues/433) problem appears during translation into scs.

Possible solutions:
1. Do checks during scs converting.
Pros:
- solves directly mentioned problem

2. Automatically change types
Cons:
- doesn't allow to do something that is not allowed in memory. But sometimes we need to do that. For example to show restricted constructions with SCg. This is a part of documentation or sometimes we need to show invalid state of KB.

3. Save identifiers with `.x => nrel_system_identifier: [];;`

4. Do nothing. Just add type specifications into scs directly and check scs sources during KB building process.
```
_x => ......;;
_x <- sc_const;;
```

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
4 possible solutions

**Comment by ValikVityaz** [comment]
Discussed solution is implemented in #431

**Pull request closed without merging**